packages/text-to-speech/src/htmlToSsml.ts

```diff
@@ -0,0 +1,178 @@
+      // Avoiding dynamic elements from being counted as anchor-allowed elements
+      ANCHOR_ELEMENTS_BLOCKED_ATTRIBUTES.some((attrib) =>
+        currentNode.hasAttribute(attrib)
+      )
+    ) {
+      continue
+    }
+
+    visitedNodeList.push(currentNode)
+    ;[].slice
+      .call(currentNode.childNodes)
+      .reverse()
+      .forEach(function (node) {
+        nodesToVisitStack.push(node)
+      })
+  }
+
+  visitedNodeList.shift()
+  visitedNodeList.forEach((node, index) => {
+    // start from index 1, index 0 reserved for anchor unknown.
+    node.setAttribute('data-omnivore-anchor-idx', (index + 1).toString())
+  })
+  return visitedNodeList
+}
+
+function emit(textItems: string[], text: string) {
+  textItems.push(text)
+}
+
+function cleanTextNode(textNode: ChildNode): String {
+  return (textNode.textContent ?? '').replace(/\s+/g, ' ')
+}
+
+function emitTextNode(textItems: string[], cleanedText: String, textNode: ChildNode) {
+  const ssmlElement = textNode.parentNode?.nodeName === 'B' ? "emphasis" : undefined
+  if (!cleanedText) { return }
+
+  if (ssmlElement) {
+    emit(textItems, `<${ssmlElement}>`)
+  }
+  emit(textItems, `${cleanedText}`)
+  if (ssmlElement) {
+    emit(textItems, `</${ssmlElement}>`)
+  }
+}
+
+function emitElement(textItems: string[], element: Element, isTopLevel: Boolean) {
+  const SKIP_TAGS = ['SCRIPT', 'STYLE', 'IMG', 'FIGURE', 'FIGCAPTION', 'IFRAME']
+  
+  const topLevelTags = ssmlTagsForTopLevelElement(element)
+  const idx = element.getAttribute('data-omnivore-anchor-idx')
+  var maxVisitedIdx = Number(idx)
+
+  if (isTopLevel) {
+    emit(textItems, topLevelTags.opening)
+  }
+
+  for (const child of Array.from(element.childNodes)) {
+    if (SKIP_TAGS.indexOf(child.nodeName) >= 0) {
+      continue
+    }
+
+    if (child.nodeType == 3 /* Node.TEXT_NODE */ && (child.textContent?.length ?? 0) > 0 ) {
+      const cleanedText = cleanTextNode(child)
+      if (cleanedText.length > 1) { // Make sure its more than just a space
+        emit(textItems, `<bookmark mark="${idx}" />`)
+      }
+      emitTextNode(textItems, cleanedText, child)
+    }
+    if (child.nodeType == 1 /* Node.ELEMENT_NODE */) {
+      maxVisitedIdx = emitElement(textItems, child as HTMLElement, false)
+    }
+  }
+
+  if (isTopLevel) {
+    emit(textItems, topLevelTags.closing)
+  }
+
+  return Number(maxVisitedIdx)
+}
+
+export type SSMLItem = {
+  open: string
+  close: string
+  textItems: string[]
+}
+
+export type VoiceOptions = {
+  primary: string
+  secondary: string
+}
+
+const startSsml = (element: Element, voices: VoiceOptions): string => {
+  const voice = element.nodeName === 'BLOCKQUOTE' ? voices.secondary : voices.primary
+  return `
+<speak xmlns="http://www.w3.org/2001/10/synthesis" xmlns:mstts="http://www.w3.org/2001/mstts" xmlns:emo="http://www.w3.org/2009/10/emotionml" version="1.0" xml:lang="en-US"><voice name="${voice}"><prosody rate="0%" pitch="0%">
```

Comment:
Maybe we should put rate=`1` or `default` according to the doc:
```Indicates the speaking rate of the text. You can express rate as:
A relative value, expressed as a number that acts as a multiplier of the default. For example, a value of 1 results in no change in the rate. A value of 0.5 results in a halving of the rate. A value of 3 results in a tripling of the rate.
A constant value:
x-slow
slow
medium
fast
x-fast
default```

Reply:
Maybe we should put pitch=`default` and some value:
```Indicates the baseline pitch for the text. You can express the pitch as:
An absolute value, expressed as a number followed by "Hz" (Hertz). For example, <prosody pitch="600Hz">some text</prosody>.
A relative value, expressed as a number preceded by "+" or "-" and followed by "Hz" or "st" that specifies an amount to change the pitch. For example: <prosody pitch="+80Hz">some text</prosody> or <prosody pitch="-2st">some text</prosody>. The "st" indicates the change unit is semitone, which is half of a tone (a half step) on the standard diatonic scale.
A constant value:
x-low
low
medium
high
x-high
default```

Reply:
We should maybe add language attribute to the `speak` element too

Reply:
`xml:lang="en-US"`

Reply:
Ah thanks, I will copy over the values you had

Reply:
I think i C&P'd this header from Azure demo app

Reply:
ok updated these ones to pass in from config. I left pitch as hard coded.